### DISCUSSION TITLE
Boards other than Teensy with virtual midi cables

### DISCUSSION THREAD

**Discussion by Roots98** [tttapa/Control-Surface#905] [General]
Is there any other board that can handle more than 4 Virtual Midi Cables other than Teensy?

**Comment by tttapa** [comment]
The main hardware requirement is sufficient memory to store the USB descriptors for the multiple cables. For most modern 32-bit chips with USB support, this is unlikely to be an issue, but IIRC, chips like the ATmega32U4 do not have sufficient memory.  
You do need to actually implement these descriptors in software, which the official MIDIUSB library does not, but you could do this yourself or try a different library.

**Reply by aberoev** [reply]
hello esp32 with adafruit tinyusb works well with this instantiation: Adafruit_USBD_MIDI usb_midi(4); (for 4 virtual midi cables) however I don't see a way to pass this through control surface's USBMIDI_Interface. Could that be possible somehow

**Reply by tttapa** [reply]
Please post your full code, and the versions of the Adafruit TinyUSB library and the ESP32 Arduino core.
